msidoc.md

```diff
@@ -0,0 +1,415 @@
+2019/05/16 21:53:58 Launching container with name: acb_step_4
+[
+  "cf2",
+  "cf3"
+]
+2019/05/16 21:54:02 Successfully executed container: acb_step_4
+2019/05/16 21:54:02 Step ID: acb_step_0 marked as successful (elapsed time in seconds: 1.070111)
+2019/05/16 21:54:02 Step ID: acb_step_1 marked as successful (elapsed time in seconds: 14.031206)
+2019/05/16 21:54:02 Step ID: acb_step_2 marked as successful (elapsed time in seconds: 7.053921)
+2019/05/16 21:54:02 Step ID: acb_step_3 marked as successful (elapsed time in seconds: 3.741742)
+2019/05/16 21:54:02 Step ID: acb_step_4 marked as successful (elapsed time in seconds: 4.274584)
+ 
+Run ID: cf3 was successful after 38s
+```
+
+## Private registry login using Managed Identities
+
+
+We will use System Identity for this example.
+
+Task:
+
+`testtask.yaml`
+``` yaml
+version: v1.0.0
+steps:
+  - build: -t {{.Values.REGISTRY1}}/hello-world:{{.Run.ID}} . -f hello-world.dockerfile
+  - push: ["{{.Values.REGISTRY1}}/hello-world:{{.Run.ID}}"]
+  - build: -t {{.Values.REGISTRY2}}/hello-world:{{.Run.ID}} . -f hello-world.dockerfile
+  - push: ["{{.Values.REGISTRY2}}/hello-world:{{.Run.ID}}"]
+```
+
+
+``` sh
+$ az acr task create -n multiple-reg -r myregistry -c https://github.com/Azure-Samples/acr-tasks.git#:multipleRegistries \
+   -f testtask.yaml --commit-trigger-enabled false --pull-request-trigger-enabled false --assign-identity
+
+{
+  "agentConfiguration": {
+    "cpu": 2
+  },
+  "creationDate": "2019-05-16T22:54:04.620200+00:00",
+  "credentials": null,
+  "id": "/subscriptions/84c559c6-30a0-417c-ba06-8a2253b388c3/resourceGroups/samashahtest-rg/providers/Microsoft.ContainerRegistry/registries/myregistry/tasks/multiple-reg",
+  "identity": {
+    "principalId": "575ec7ae-e21e-402d-84ee-a12fd5b20a47",
+    "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+    "type": "SystemAssigned",
+    "userAssignedIdentities": null
+  },
+  "location": "westus",
+  "name": "multiple-reg",
+  "platform": {
+    "architecture": "amd64",
+    "os": "linux",
+    "variant": null
+  },
+  "provisioningState": "Succeeded",
+  "resourceGroup": "samashahtest-rg",
+  "status": "Enabled",
+  "step": {
+    "baseImageDependencies": null,
+    "contextAccessToken": null,
+    "contextPath": "https://github.com/Azure-Samples/acr-tasks.git#:multipleRegistries",
+    "taskFilePath": "testtask.yaml",
+    "type": "FileTask",
+    "values": [],
+    "valuesFilePath": null
+  },
+  "tags": null,
+  "timeout": 3600,
+  "trigger": {
+    "baseImageTrigger": {
+      "baseImageTriggerType": "Runtime",
+      "name": "defaultBaseimageTriggerName",
+      "status": "Enabled"
+    },
+    "sourceTriggers": null,
+    "timerTriggers": null
+  },
+  "type": "Microsoft.ContainerRegistry/registries/tasks"
+}
+
+// Capture id and principal_id from response.
+
+$ az acr task credential add -n multiple-reg -r myregistry --login-server customregistry1.azurecr.io --use-identity $principal_id
+{
```

Comment:
To use system assigned MSI, the command I needed to use looked like (with the `[system]`:

```
az acr task credential add -n multiple-reg -r myregistry --login-server customregistry1.azurecr.io --use-identity [system]
```

Reply:
good catch